scripts/tests_like_ci/rust_tests.sh

```diff
@@ -0,0 +1,113 @@
+    set -x
+    eval $command
+}
+
+function test_rust_crates() {
+    # Tests written with #[sim_test] are often flaky if run as #[tokio::test] - this var
+    # causes #[sim_test] to only run under the deterministic `simtest` job, and not the
+    # non-deterministic `test` job.
+    export IOTA_SKIP_SIMTESTS=1
+    cargo nextest run --config-file .config/nextest.toml --profile ci
+}
+
+function test_external_crates() {
+    cargo nextest run --config-file .config/nextest.toml --manifest-path external-crates/move/Cargo.toml -E '!test(prove) and !test(run_all::simple_build_with_docs/args.txt) and !test(run_test::nested_deps_bad_parent/Move.toml)' --profile ci
+}
+
+
+
+function check_unused_deps() {
+    cargo +nightly ci-udeps --all-features
+    cargo +nightly ci-udeps --no-default-features
+}
+
+function test_extra() {
+    export IOTA_SKIP_SIMTESTS=1
+    cargo run --package iota-benchmark --bin stress -- --log-path ${WD}/.cache/stress.log --num-client-threads 10 --num-server-threads 24 --num-transfer-accounts 2 bench --target-qps 100 --num-workers 10  --transfer-object 50 --shared-counter 50 --run-duration 10s --stress-stat-collection
+    cargo test --doc
+    cargo doc --all-features --workspace --no-deps
+    ${WD}/scripts/execution_layer.py generate-lib;
+    ${WD}/scripts/changed-files.sh;
+}
+
+function simtests() {
+    export MSIM_WATCHDOG_TIMEOUT_MS=60000
+    scripts/simtest/cargo-simtest simtest --profile ci --color always
+    scripts/simtest/stress-new-tests.sh
+}
+
+function tests_using_postgres() {
+    restart_postgres
+    cargo nextest run --no-fail-fast --test-threads 1 --package iota-graphql-rpc --test e2e_tests --test examples_validation_tests --features pg_integration
+    cargo nextest run --no-fail-fast --test-threads 1 --package iota-graphql-rpc --lib --features pg_integration -- test_query_cost
+    cargo nextest run --no-fail-fast --test-threads 8 --package iota-graphql-e2e-tests --features pg_integration
+    cargo nextest run --no-fail-fast --test-threads 1 --package iota-cluster-test --test local_cluster_test --features pg_integration
+    cargo nextest run --no-fail-fast --test-threads 1 --package iota-indexer --test ingestion_tests --features pg_integration
+    # Iota-indexer's RPC tests, which depend on a shared runtime, are incompatible with nextest due to its process-per-test execution model.
+    # cargo test, on the other hand, allows tests to share state and resources by default.
+    cargo test --profile simulator --package iota-indexer --test rpc-tests --features shared_test_runtime
+}
+
+
+# Running all the tests will compile different sets of crates and take a lot of storage (>500GB)
+# If your machine has less storage, you can run only part of the tests (at a time), 
+# use the name of the function to run as a subcommand, for instance:
+# ./scripts/tests_like_ci/rust_tests.sh simtests
+if [ -n "$RUN_ONLY_STEP" ]; then
+    if declare -f "$RUN_ONLY_STEP" > /dev/null; then
+        "$RUN_ONLY_STEP"
```

Comment:
will this allow to execute arbitrary commands?

e.g. `RUN_ONLY_STEP=rm -Rf /`

Reply:
it would. Though it's intended for the user only, to type the step name
I could replace it with a `CASE $1 in` and list accepted values explicitly ?

Reply:
I would prefer that, yes.